kafka-connect/kafka-connect/src/test/java/org/apache/iceberg/connect/data/RecordConverterTest.java

```diff
@@ -921,11 +948,26 @@ private void assertRecordValues(Record record) {
     assertThat(rec.getField("dec")).isEqualTo(DEC_VAL);
     assertThat(rec.getField("s")).isEqualTo(STR_VAL);
     assertThat(rec.getField("b")).isEqualTo(true);
-    assertThat(rec.getField("u")).isEqualTo(UUID_VAL);
-    assertThat(rec.getField("f")).isEqualTo(BYTES_VAL);
+    assertThat(rec.getField("f")).isEqualTo(BYTES_VAL.array());
     assertThat(rec.getField("bi")).isEqualTo(BYTES_VAL);
     assertThat(rec.getField("li")).isEqualTo(LIST_VAL);
     assertThat(rec.getField("ma")).isEqualTo(MAP_VAL);
+
+    IcebergSinkConfig icebergSinkConfig = mock(IcebergSinkConfig.class);
+    when(icebergSinkConfig.tableConfig(any())).thenReturn(mock(TableSinkConfig.class));
+    when(icebergSinkConfig.writeProps())
+        .thenReturn(ImmutableMap.of(TableProperties.DEFAULT_FILE_FORMAT, format));
+    WriteResult result =
+        writeTest(ImmutableList.of(rec), icebergSinkConfig, UnpartitionedWriter.class);
+
+    if (format.equals(FileFormat.PARQUET.name().toLowerCase(Locale.ROOT))) {
+      assertThat(rec.getField("u")).isEqualTo(UUIDUtil.convert(UUID_VAL));
```

Comment:
In generally I don't like replicating the implementation logic in the test. Can we check this in a more agnostic way?

Reply:
make sense ! Addressed the feedback.